src/ucs/datastruct/ptr_array.c

```diff
@@ -202,12 +203,52 @@ unsigned ucs_ptr_array_insert(ucs_ptr_array_t *ptr_array, void *value)
     return element_index;
 }
 
+unsigned ucs_ptr_array_alloc(ucs_ptr_array_t *ptr_array, unsigned element_count)
```

Comment:
1. maybe ucs_ptr_array_bulk_alloc?
2. maybe implement ucs_ptr_array_insert() on top of ucs_ptr_array_alloc(ptr_array, element_count=1) ?